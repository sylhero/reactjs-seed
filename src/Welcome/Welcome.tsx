/** @jsx jsx */
import React from 'react';

import { jsx, css } from '@emotion/react';

const base = css`
    color: hotpink;
`;
const welcomeStyle = css`
    ${base}
    background-color: blue;
`;

const Welcome = (): React.ReactElement => {
    return <div css={welcomeStyle}>welcome</div>;
};

export default Welcome;
