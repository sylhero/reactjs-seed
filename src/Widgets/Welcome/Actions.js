import ActionTypes from './ActionTypes';
import Apis from './Apis';

const getData = () => {
    return (dispatch) => {
        dispatch({
            type: ActionTypes.GET_TEST_DATA,
            payload: Apis.getTestData
        });
    };
};

const updateData = () => {
    return (dispatch) => {
        dispatch({
            type: ActionTypes.UPDATE_TEST_DATA,
            payload: {
                data: {
                    title: 'woru'
                }
            }
        });
    };
};

const Actions = {
    getData,
    updateData
};

export default Actions;
