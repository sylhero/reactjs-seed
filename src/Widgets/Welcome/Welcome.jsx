// @flow strict
import React from 'react';
import { Icon } from 'antd';

import WelcomeCss from './Welcome.scss';

type Props = {
    username: string
};

const Welcome = (props: Props) => {
    return (
        <div>
            <Icon type="smile-o" />
            <h1 className={WelcomeCss.test}>
                {props.username}
            </h1>
        </div>
    );
};

export default Welcome;
