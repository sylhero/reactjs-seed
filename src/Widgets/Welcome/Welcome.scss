/* for variable please see src/_constant.scss */

.test {
    color: $my-pink;
}
