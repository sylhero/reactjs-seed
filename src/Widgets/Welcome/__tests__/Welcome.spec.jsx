import React from 'react';
import { shallow } from 'enzyme';
import Welcome from '../Welcome';


/*
    shallow --- only render one level deep of the component
    mount --- full rendering (rendering in a browser-like environment with jsdom)
    render --- render the component into html

    https://github.com/airbnb/enzyme/tree/master/docs/api
 */
describe('Welcome', () => {
    it('should create a snapshot', () => {
        const props = {
            username: 'test'
        };
        const wrapper = shallow(<Welcome {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
