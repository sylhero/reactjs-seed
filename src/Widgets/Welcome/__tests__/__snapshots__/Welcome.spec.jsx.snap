// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should create a snapshot 1`] = `
<div>
  <Icon
    type="smile-o"
  />
  <h1>
    test
  </h1>
</div>
`;
